atomic_reactor/plugins/pre_fetch_maven_artifacts.py

```diff
@@ -185,16 +235,22 @@ def run(self):
         pnc_requests = util.read_fetch_artifacts_pnc(self.workflow) or {}
         url_requests = util.read_fetch_artifacts_url(self.workflow) or []
 
-        download_queue = (self.process_by_nvr(nvr_requests) +
-                          self.process_pnc_requests(pnc_requests) +
-                          self.process_by_url(url_requests))
+        components, nvr_download_queue = self.process_by_nvr(nvr_requests)
+        url_download_queue, source_download_queue = self.process_by_url(url_requests)
+        pnc_artifact_ids, pnc_download_queue, pnc_build_metadata = self.process_pnc_requests(
+            pnc_requests)
+
+        download_queue = pnc_download_queue + nvr_download_queue + url_download_queue
 
         download_to_build_dir = functools.partial(self.download_files, download_queue)
         self.workflow.build_dir.for_all_platforms_copy(download_to_build_dir)
 
-        pnc_artifact_ids = self.get_pnc_artifact_ids(pnc_requests)
-
         return {
+            'components': components,
             'download_queue': [dataclasses.asdict(download) for download in download_queue],
+            'no_source': self.no_source_artifacts,
             'pnc_artifact_ids': pnc_artifact_ids,
+            'pnc_build_metadata': pnc_build_metadata,
+            'source_download_queue': source_download_queue,
+            'source_url_to_artifacts': self.source_url_to_artifacts,
```

Comment:
Note that plugin results have to be convertible to JSON now (for context sharing). I'm not sure what the types of these values are, but I think at least `source_download_queue` is not JSON compatible

Reply:
added serialization and deserialization